### PR TITLE
New. SFW subnets counter.

### PR DESCRIPTION
**Pull request by alexandergull**

### CODE REVIEW

File: lib/Cleantalk/Common/Firewall/FirewallUpdater.php [hunk 1]
```diff
@@ -245,8 +257,20 @@ public function update()
...
     {
         $api = $this->api;
+
         $result = $api::method__get_2s_blacklists_db( $api_key, 'multifiles', '3_0' );
         sleep(4);
+
+        return $result;
+    }
+
+    private function getSfwBlacklistsCount( $api_key )
```
Comment (ArtemAnoshin) [inline lib/Cleantalk/Common/Firewall/FirewallUpdater.php:267]: Maybe to take out to another class more suitable?

File: lib/Cleantalk/Common/Firewall/FirewallUpdater.php [hunk 2]
```diff
@@ -165,13 +167,18 @@ public function update()
 
                     // Do writing to the DB
                     reset( $lines );
+                    $subnets_count_from_stats = isset($helper::getFwStats()['firewall_subnets_count'])
+                        ? $helper::getFwStats()['firewall_subnets_count']
+                        : 0;
+                    $subnets_count = $subnets_count_from_stats ?: 0;
```
Comment (Glomberg) [inline lib/Cleantalk/Common/Firewall/FirewallUpdater.php:173]: It seems the variable `$subnets_count_from`_stats is redundant

File: lib/Cleantalk/Common/Firewall/FirewallUpdater.php [hunk 3]
```diff
@@ -165,13 +167,18 @@ public function update()
 
                     // Do writing to the DB
                     reset( $lines );
+                    $subnets_count_from_stats = isset($helper::getFwStats()['firewall_subnets_count'])
```
Comment (Glomberg) [inline lib/Cleantalk/Common/Firewall/FirewallUpdater.php:170]: The condition `isset($helper::getFwStats()['firewall_subnets_count'])` is not enough. Need to compare on 0 too: `$helper::getFwStats()['firewall_subnets_count'] > 0`

File: lib/Cleantalk/Common/Firewall/FirewallUpdater.php [hunk 4]
```diff
@@ -215,6 +224,9 @@ public function update()
                         //Files array is empty update sfw stats
                         $helper::SfwUpdate_DoFinisnAction();
 
+                        if ( $fw_stats['firewall_subnets_count'] == $this->getSfwBlacklistsCount($this->api_key) ) {
```
Comment (Glomberg) [inline lib/Cleantalk/Common/Firewall/FirewallUpdater.php:227]: This additional call of the API method `get_2s_blacklists_db` need to be removed!

**Review by Glomberg — CHANGES_REQUESTED** [review]

File: lib/Cleantalk/Common/Firewall/FirewallUpdater.php (hunk 1 above)
Comment (ArtemAnoshin) [reply lib/Cleantalk/Common/Firewall/FirewallUpdater.php:267]: All good

File: lib/Cleantalk/Common/Firewall/FirewallUpdater.php (hunk 2 above)
Comment (ArtemAnoshin) [reply lib/Cleantalk/Common/Firewall/FirewallUpdater.php:173]: Yes, fixed

File: lib/Cleantalk/Common/Firewall/FirewallUpdater.php (hunk 3 above)
Comment (ArtemAnoshin) [reply lib/Cleantalk/Common/Firewall/FirewallUpdater.php:170]: Fixed

**Comment by alexandergull** [comment]
We will do common lib and queue for SFW update. Close this one.

**Pull request closed without merging**